forem/forem
# Feeds::Import: only import feeds that haven't been recently updated

## What type of PR is this? (check all applicable)


## Description

This PR set at 4 hours the time span to ignore recently updated feeds. By moving the `feed_fetched_at` update to the end of the entire process (fetching from the web, parsing and importing articles) we should be safely confident in ignoring those users whose `feed_fetched_at` has been updated in the past 4 hours.

Thanks @mstruve for suggesting this approach.

## QA Instructions, Screenshots, Recordings

1. Populate `feed_url` for at least one user, eg. `u = User.last; u.update(feed_url: "https://medium.com/feed/@vaidehijoshi")`
2. Start Sidekiq in a tab, `bundle exec sidekiq`
3. Run `Feeds::ImportArticlesWorker.perform_async(nil, 4.hours.ago)` manually or enable `:feeds_import` feature flag and run `Articles::RssReaderWorker.perform_async`
4. Verify that articles have been imported
5. Run step 3 again, check Sidekiq logs to see that the user has been ignored
6. Re-run  `Feeds::ImportArticlesWorker.perform_async(nil, 1.second.ago)` and check the user is picked up again (by lowering the time, we make sure the logic works at the end of the interval)

## Added tests?

      have not been included_
